### PR TITLE
[api-extractor] Introduce sort modifier to api extractor and dts rollup to avoid spurious reports between builds

### PR DESCRIPTION
**Pull request by ajithAPER**
**Summary**
Fixed https://github.com/microsoft/rushstack/issues/1958

**Details**
API reports and DTS rollouts generated show signs of spurious inferred union types between incremental build. This implementation simply introduces a sort modifier to sort all union type in DTS rollups and API reports.

**How was it tested**
Tested against reports within the repository generated automatically.

### CODE REVIEW

**Comment by ajithAPER** [comment]
This pull request cleans up the additional changes noted in https://github.com/microsoft/rushstack/pull/3698 and reintriduces the original change.

File: common/changes/@microsoft/api-extractor/api-extractor-sort-modifier-for-union-spans-in-generators_2023-01-09-04-30.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-extractor",
+      "comment": "Introduces sort modifier in ApiReportGenerator and DtsRollupGenerator for union types to avoid spurious outputs between builds",
```
Comment (iclanton) [inline common/changes/@microsoft/api-extractor/api-extractor-sort-modifier-for-union-spans-in-generators_2023-01-09-04-30.json:5]:
```suggestion
      "comment": "Make the ordering of elements in union types in API reports and .D.TS rollups always be in alphabetical order.",
```

File: common/changes/@microsoft/api-extractor/api-extractor-sort-modifier-for-union-spans-in-generators_2023-01-09-04-30.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+      "type": "patch"
```
Comment (iclanton) [inline common/changes/@microsoft/api-extractor/api-extractor-sort-modifier-for-union-spans-in-generators_2023-01-09-04-30.json:6]:
```suggestion
      "type": "minor"
```

**Comment by iclanton** [comment]
As mentioned in the original issue, it would be nice to preserve the order of elements in explicitly defined union types and only apply the sorting to inferred types. It looks like this change sorts all union types. How difficult would it be to only sort inferred types? If it's too difficult to do that, can this be behind a configuration option?

Imagine you have a type that looks like this in source:

```ts
type DigitAsWord = 'zero' | 'one' | 'two' | 'three' | 'four' | 'five' | 'six' | 'seven' | 'eight' | 'nine';
```

where the constituents of the type have a natural order. If they're always sorted, this type will look like this in the API report:

```ts
type DigitAsWord = 'eight' | 'five' | 'four' | 'nine' | 'one' | 'seven' | 'six' | 'two' | 'zero';
```

which is harder to review for correctness. Did you notice that `'three'` is missing from between `'six'` and `'two'` in the code snippet above? That would've been much easier to catch if the elements were in the same order as the source code.

**Pull request closed without merging**